pdf-2/src/main/java/com/baeldung/exceltopdf/ExcelToPDFConverter.java

```diff
@@ -0,0 +1,167 @@
+import org.apache.poi.xssf.usermodel.XSSFSheet;
+import org.apache.poi.xssf.usermodel.XSSFWorkbook;
+import org.apache.poi.ss.usermodel.CellType;
+import org.apache.poi.ss.usermodel.Row;
+import org.apache.poi.ss.usermodel.Cell;
+import com.itextpdf.text.BaseColor;
+import com.itextpdf.text.Document;
+import com.itextpdf.text.DocumentException;
+import com.itextpdf.text.Element;
+import com.itextpdf.text.Font;
+import com.itextpdf.text.Paragraph;
+import com.itextpdf.text.Phrase;
+import com.itextpdf.text.pdf.PdfPCell;
+import com.itextpdf.text.pdf.PdfPTable;
+import com.itextpdf.text.pdf.PdfWriter;
+
+public class ExcelToPDFConverter {
+
+    public static XSSFWorkbook readExcelFile(String excelFilePath) throws IOException {
+        FileInputStream inputStream = new FileInputStream(excelFilePath);
+        XSSFWorkbook workbook = new XSSFWorkbook(inputStream);
+        inputStream.close();
+        return workbook;
+    }
+    private static Document createPDFDocument(String pdfFilePath) throws IOException, DocumentException {
+        Document document = new Document();
+        PdfWriter.getInstance(document, new FileOutputStream(pdfFilePath));
+        document.open();
+        return document;
+    }
+    public static void convertExcelToPDF(String excelFilePath, String pdfFilePath) throws IOException, DocumentException {
+        XSSFWorkbook workbook = readExcelFile(excelFilePath);
+        Document document = createPDFDocument(pdfFilePath);
+
+        for (int i = 0; i < workbook.getNumberOfSheets(); i++) {
+            XSSFSheet worksheet = workbook.getSheetAt(i);
+
+            // Add header with sheet name as title
+            Paragraph title = new Paragraph(worksheet.getSheetName(), new Font(Font.FontFamily.HELVETICA, 18, Font.BOLD));
+            title.setAlignment(Element.ALIGN_CENTER);
+            document.add(title);
+
+            createAndAddTable(worksheet, document);
+            // Add a new page for each sheet (except the last one)
+            if (i < workbook.getNumberOfSheets() - 1) {
+                document.newPage();
+            }
+        }
+
+        document.close();
+        workbook.close();
+    }
+
+    private static void createAndAddTable(XSSFSheet worksheet, Document document) throws DocumentException, IOException {
+        PdfPTable table = new PdfPTable(worksheet.getRow(0).getPhysicalNumberOfCells());
+        table.setWidthPercentage(100);
+        addTableHeader(worksheet, table);
+        addTableData(worksheet, table);
+        document.add(table);
+    }
+
+    private static void addTableHeader(XSSFSheet worksheet, PdfPTable table) throws DocumentException, IOException {
+        Row headerRow = worksheet.getRow(0);
+        for (int i = 0; i < headerRow.getPhysicalNumberOfCells(); i++) {
+            Cell cell = headerRow.getCell(i);
+            String headerText = cell.getCellType() == CellType.STRING ? cell.getStringCellValue() : String.valueOf(cell.getNumericCellValue());
```

Comment:
Does this work if the cell type isn't String or Numeric, for example Blank?

Reply:
Thanks for pointing this out, will fix it to handle blank and other types.